lib/curl_fnmatch.c

```diff
@@ -71,18 +71,19 @@ typedef enum {
 #define SETCHARSET_OK     1
 #define SETCHARSET_FAIL   0
 
-static int parsekeyword(unsigned char **pattern, unsigned char *charset)
+static int parsekeyword(char **pattern, char *charset)
 {
   parsekey_state state = CURLFNM_PKW_INIT;
 #define KEYLEN 10
   char keyword[KEYLEN] = { 0 };
-  int found = FALSE;
-  int i;
-  unsigned char *p = *pattern;
+  bool found = FALSE;
+  unsigned int i;
+  char *p = *pattern;
+  char c;
```

Comment:
Why the changed scope for `c` here?

Reply:
c is defined elsewhere, causing shadowing